src/checker.rs

```diff
@@ -690,6 +697,128 @@ pub(crate) enum CheckerInst {
     },
 }
 
+#[derive(Debug)]
+struct DebugLocationEntry {
+    vreg: VReg,
+    alloc: Allocation,
+    label: u32,
+}
+
+#[derive(Debug)]
+struct DebugLocations {
+    expected_vreg_locations: FxHashMap<(VReg, (ProgPoint, ProgPoint), u32), Allocation>,
+}
+
+impl DebugLocations {
+    fn ranges_overlaps(
+        (start_point, end_point): (ProgPoint, ProgPoint),
+        (start_inst, end_inst): (Inst, Inst),
+    ) -> Option<(ProgPoint, ProgPoint)> {
+        if end_inst <= start_point.inst() || start_inst >= end_point.inst() {
+            None
+        } else {
+            let point0 = if start_point.inst() >= start_inst {
```

Comment:
Similar here to below -- `std::cmp::min` / `std::cmp::max`?

Reply:
And can we name this `intersects`, and factor it out to a common method on `ProgPoint` shared by the other use below? I see the args are slightly different -- ProgPoint/Inst here, Inst/ProgPoint below, but unless there are subtleties around the handling of begin/end, we could take ProgPoints for both ranges and convert Insts to ProgPoints as needed at callsites I think?